### DISCUSSION TITLE
Allows using Astro.currentLocale in components to access the current language

### DISCUSSION THREAD

**Discussion by LuHugo** [withastro/starlight#1176] [Feature Requests]
### What version of `starlight` are you using?

0.14.0

### What is your idea?

i18n has been configured in the starlight configuration, but `Astro.currentLocale` cannot be used in the custom component to obtain the current language. The value returned by `Astro.currentLocale` is `undefined`. Is it possible to get the language through `Astro.currentLocale` or should there be a dedicated way to get the currently selected language?

### Why is this feature necessary?

If there are some URLs in the component that need to be added with the current language prefix, you need to get the current language.

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by astridx** [comment]
I think you can use `Astro.props.lang` instead of `Astro.currentLocale`. It works in my custom component.

Please correct me if this is not the recommended way. I have seen it in the code. For example here: 

https://github.com/withastro/starlight/blob/a0f6fa80d9be1fbe85530323a0b2523acb83a15a/packages/starlight/components/Page.astro#L42

**Reply by delucis** [reply]
Yes, this works well in component overrides! But it does not work in components inside content which don’t get passed those props. Instead you currently need to manually extract the locale from `Astro.url`.

I’m +1 on refactoring to support `Astro.currentLocale`. Will try to take a look soon unless someone is interested in doing it themselves!

**Reply by lorenzolewis** [reply]
This would be amazing to have access to. I think with this in place then filling in the missing pieces with the now extensible i18n data collection would solve some low-hanging fruit for when a full i18n solution isn't required. It may also be useful to know if the page being rendered is a fallback page or not.

**Comment by delucis** [comment]
This is supported as of Starlight v0.24 thanks to https://github.com/withastro/starlight/pull/1841!
